### PR TITLE
[CI] Add a C++11 build

### LINKED ISSUE #2151: [CI] Add a C++11 build
Before upgrading CI to C++14, add a dedicated build for C++11, so that breaks affecting C++11 only can be detected.

This build will use the last third party libraries still available for C++11.

There is no need to duplicate every job for C++11 / C++14, so only one C++11 job with as many features enabled as possible should be sufficient.

### PR DESCRIPTION
**Pull request by marcalff**
Fixes #2151

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

**Review by owent — APPROVED** [review]
LGTM and thanks

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -29,6 +29,44 @@ jobs:
...
+      run: |
+        sudo -E ./ci/setup_cmake.sh
+        sudo -E ./ci/setup_ci_environment.sh
+        sudo -E ./ci/install_protobuf.sh
+    - name: run cmake gcc (maintainer mode, async)
+      env:
+        CC: /usr/bin/gcc-12
+        CXX: /usr/bin/g++-12
+      run: |
+        sudo -E ./ci/setup_thrift.sh
```
Comment (esigo) [inline .github/workflows/ci.yml:58]: Shouldn't we already remove this, as Jaeger is going to be removed soon?

Comment (marcalff) [reply .github/workflows/ci.yml:58]: Indeed, thanks. Will fix tomorrow

**Review by esigo — APPROVED** [review]
LGTM
thanks :)

**Pull request merged**